src/class-wc-gateway-btcpay.php

```diff
@@ -735,8 +736,18 @@ public function process_payment($order_id)
                 $this->log('    [Info] Invoice object created successfully...');
             }
 
-            $order_number = $order->get_order_number();
-            $invoice->setOrderId((string)$order_number);
+            $order_url = $order->get_edit_order_url();
+
+            $pos_data = array(
```

Comment:
Nice idea adding that metadata here for easier handling.

Reply:
Nested the metadata as proposed by @kukks (see a2bf31f6c8f6f6245bca0956390b33b7c55bb4a6)